frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx

```diff
@@ -1,32 +1,46 @@
 import * as React from 'react';
-import { Perspective, PerspectiveContext } from '@console/dynamic-plugin-sdk';
-import { usePerspectives } from '@console/shared/src';
+import { useRouteMatch } from 'react-router-dom';
+import { PerspectiveContext } from '@console/dynamic-plugin-sdk';
+import { CLUSTER_SCOPED_PREFIXES } from '@console/internal/components/utils';
+import { usePerspectives, useQueryParams } from '@console/shared/src';
 import PerspectiveDetector from './PerspectiveDetector';
 import { useValuesForPerspectiveContext } from './useValuesForPerspectiveContext';
 
 type DetectPerspectiveProps = {
   children: React.ReactNode;
 };
 
-const getPerspectiveURLParam = (perspectives: Perspective[]) => {
-  const perspectiveIDs = perspectives.map(
-    (nextPerspective: Perspective) => nextPerspective.properties.id,
-  );
+const useOverridePerspective = (activePerspective) => {
+  const perspectiveExtensions = usePerspectives();
+  const perspectiveParam = useQueryParams().get('perspective');
```

Comment:
the `frontend` TC are failing due to typescript error:
```
TypeError: src_1.useQueryParams is not a function
```

```suggestion
  const queryParams = useQueryParams()
  const perspective = queryParams.get('perspective');
```

Reply:
@TheRealJon so the issue is still in the CI:

<img width="948" alt="Screenshot 2022-11-28 at 17 22 05" src="https://user-images.githubusercontent.com/1668218/204328759-12d228c1-0270-4fc6-b167-c7ed4bfc9b45.png">